### PR TITLE
Use GL_ARB_separate_shader_objects in compat GLSL

### PR DESCRIPTION
**Pull request by dsrbecky**
Try to enable the extension if it is available

### CODE REVIEW

File: gapis/shadertools/cc/libmanager.cpp [hunk 1]
```diff
@@ -228,6 +228,15 @@ code_with_debug_info_t* convertGlsl(const char* input, size_t length, const opti
 
   std::string source = spirv2glsl(std::move(spirv_new));
 
+  // Add support for layout(location) as needed by separate shader objects
+  // TODO: Postprocessing source code like this is fairly hacky. Find a better way.
+  auto extensionPos = source.find("#ifdef GL_ARB_shading_language_420pack");
```
Comment (ben-clayton) [inline gapis/shadertools/cc/libmanager.cpp]: This is a bit gross. Maybe we should use a regex to at least handle more than 1 whitespace between the #ifdef and the extension name?

**Comment by ben-clayton** [comment]
Did you mean to close this?

**Comment by ben-clayton** [comment]
Oh, you emailed me. Shame.

**Comment by dsrbecky** [comment]
Well, glslang check still warns us the the code we produce is invalid.
After a bit of spec reading, I think it might be right.
Although it does seem to fix it on my computer.... I think.... "gapit report" is moody today.
I would be still curious to know if this PR fixes it for you.

**Comment by ben-clayton** [comment]
> I would be still curious to know if this PR fixes it for you.

Afraid not, I still get the same issues appearing.

**Comment by dsrbecky** [comment]
Thought so, pity.
I will see if just removing the locations works.
Does gapit report show the shader compilation issues for you?

**Pull request closed without merging**